test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs

```diff
@@ -1924,7 +1924,7 @@ public static TheoryData<CreateTokenTheoryData> RoundTripJWEKeyWrapTestCases
                         Payload = Default.PayloadString,
                         SigningCredentials = Default.SymmetricSigningCredentials,
                         EncryptingCredentials = new EncryptingCredentials(Default.SymmetricEncryptionKey256, SecurityAlgorithms.Aes128KW, SecurityAlgorithms.Aes128CbcHmacSha256),
-                        ExpectedException = ExpectedException.ArgumentOutOfRangeException("IDX10662:")
+                        ExpectedException = ExpectedException.SecurityTokenKeyWrapException("IDX10662:")
```

Comment:
Something doesn't add up here. Only these two lines are throwing exceptions with IDX10662, and both of them are ArgumentOutOfRangeExceptions.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/1a1ff1f75bf042451c136e1246690c036a37a4d6/src/Microsoft.IdentityModel.Tokens/Encryption/SymmetricKeyWrapProvider.cs#L343

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/1a1ff1f75bf042451c136e1246690c036a37a4d6/src/Microsoft.IdentityModel.Tokens/Encryption/SymmetricKeyWrapProvider.cs#L351

Reply:
I would have change this to include the inner exception, but the exception code is using the exception as a message instead of as an inner exception, this is coming from the wrap or unwrap, since the validation is deferred 

So something like this is going through the deferred validation code then catching and setting the exception message.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/1a1ff1f75bf042451c136e1246690c036a37a4d6/src/Microsoft.IdentityModel.Tokens/Encryption/SymmetricKeyWrapProvider.cs#L238

Reply:
Oh, I see, e.g. SecurityTokenKeyWrapException IDX10658 was thrown, and you are checking for the exception type and id of the original message (IDX10662).